src/python/pants/engine/target.py

```diff
@@ -612,9 +612,13 @@ class TransitiveTargetsRequest:
     """
 
     roots: Tuple[Address, ...]
+    include_special_cased_deps: bool
```

Comment:
Ok, forget the above -- I tried to implement a new dataclass and it was too hard.

Could we make this an `Enum` instance instead?

Reply:
What would the `Enum` be? There are only two variants: include or exclude. 

While I generally like Enums for modeling (especially over strings), they add new complexity. Unless we believe there are going to be additional states beyond this binary in the future, an enum seems like unnecessary complexity imo, such as adding yet another class you need to import from `target.py`, which is already gargantuan.

Reply:
> What would the Enum be? There are only two variants: include or exclude.

You said in this comment https://github.com/pantsbuild/pants/pull/10923#discussion_r501942411:
> There is very little unifying about this parent class, outside of a) these are like the normal dependencies field, and b) you are going to do something magical with the values

That seems to me to imply we are going to want to do many other things with this field in the future. An enum allows for that. boolean fields are often a code smell: see https://softwareengineering.stackexchange.com/a/148058.

> such as adding yet another class you need to import from target.py, which is already gargantuan.

I think this is necessary complexity, because each element is individually cacheable. When I responded to a user in the slack yesterday who mentioned that it was a little off-putting to import so many things by mentioning how the point is to make a 100% cacheable and performant build tool, they seemed to accept that very easily.

I think if this is a concern, we could consider methods to bring elements into scope automatically, perhaps by linking classes to other classes. #10924 generalizes the `inject_for`/`infer_from` pattern with a `ConvertibleFrom` mixin, which makes a lot of the small machinery fall away, for example.

Reply:
> boolean fields are often a code smell:

Sure, but premature abstraction is also a code smell.

Reply:
I do not think that giving each choice of an option a name is "premature abstraction" and I'm confused as to why you've decided this is the case given that it was decided a whole separate issue needed to be created to figure out how to handle this problem of the kind of dependencies to retrieve. It's clearly not a trivial distinction. I don't care enough to continue this discussion.